projects/packages/videopress/src/client/admin/components/video-details-card/index.tsx

```diff
@@ -12,70 +12,98 @@ import classnames from 'classnames';
  */
 import ClipboardButtonInput from '../clipboard-button-input';
 import styles from './style.module.scss';
-import { VideoDetailsProps, VideoThumbnailEditProps } from './types';
+import { VideoDetailsProps, VideoThumbnailProps, VideoThumbnailDropdownProps } from './types';
 import type React from 'react';
 
+export const VideoThumbnailDropdown: React.FC< VideoThumbnailDropdownProps > = ( {
+	onUseDefaultThumbnail,
+	onSelectFromVideo,
+	onUploadImage,
+} ) => {
+	return (
+		<div className={ styles[ 'video-thumbnail-edit' ] }>
+			<Dropdown
+				position="bottom left"
+				renderToggle={ ( { isOpen, onToggle } ) => (
+					<Button
+						variant="secondary"
+						className={ styles[ 'thumbnail__edit-button' ] }
+						icon={ edit }
+						onClick={ onToggle }
+						aria-expanded={ isOpen }
+					/>
+				) }
+				renderContent={ () => (
+					<>
+						<Button
+							weight="regular"
+							fullWidth
+							variant="tertiary"
+							icon={ image }
+							onClick={ onUseDefaultThumbnail }
+						>
+							{ __( 'Use default thumbnail', 'jetpack-videopress-pkg' ) }
+						</Button>
+						<Button
+							weight="regular"
+							fullWidth
+							variant="tertiary"
+							icon={ media }
+							onClick={ onSelectFromVideo }
+						>
+							{ __( 'Select from video', 'jetpack-videopress-pkg' ) }
+						</Button>
+						<Button
+							weight="regular"
+							fullWidth
+							variant="tertiary"
+							icon={ cloud }
+							onClick={ onUploadImage }
+						>
+							{ __( 'Upload image', 'jetpack-videopress-pkg' ) }
+						</Button>
+					</>
+				) }
+			/>
+		</div>
+	);
+};
+
 /**
  * React component to display video thumbnail.
  *
- * @param {VideoThumbnailEditProps} props - Component props.
- * @returns {React.ReactNode} - VideoThumbnailEdit react component.
+ * @param {VideoThumbnailProps} props - Component props.
+ * @returns {React.ReactNode} - VideoThumbnail react component.
  */
-export const VideoThumbnailEdit: React.FC< VideoThumbnailEditProps > = ( {
+export const VideoThumbnail: React.FC< VideoThumbnailProps > = ( {
```

Comment:
Do you have plans to extract this to his own file?

Reply:
maybe later?

Reply:
Yep, just to know if you have that in mind.